mm2src/mm2_test_helpers/dummy_files/iris_test_history.json

```diff
@@ -0,0 +1,170 @@
+[
+    {
+        "tx_hex": "0a2a696161317a36303967397a386566346a6c6b63666767333663326c6a67636d6e6135306c32796a7261301240384636304534433336384335344539343233344145424135394331344531384146413138443437303530314434383944444641364539363635444135323738451a4064363034613132626531616239643966343531306435343832666232323762613530363238373531326466333561626337303730313763366439306165303837",
+        "tx_hash": "6BA67D4C7A0F48E5D85EE18EF5A649ED456B539D902828D56849B57DCFF5914B",
+        "from": [
+            "iaa1z609g9z8ef4jlkcfgg36c2ljgcmna50l2yjra0"
+        ],
+        "to": [
+            "iaa1erfnkjsmalkwtvj44qnfr2drfzdt4n9ldh0kjv"
+        ],
+        "total_amount": "0.026427",
+        "spent_by_me": "0.026427",
+        "received_by_me": "0",
+        "my_balance_change": "-0.026427",
+        "block_height": 6148319,
+        "timestamp": 0,
+        "fee_details": {
+            "type": "Utxo",
```

Comment:
This is a bug which should show up for every non-utxo coins. In current db function enum deserialization from database is not working as expected.

Change the order of `Utxo` like:
```diff
--- a/mm2src/coins/lp_coins.rs
+++ b/mm2src/coins/lp_coins.rs
@@ -963,13 +963,13 @@ pub struct VerificationResponse {
 #[derive(Clone, Debug, PartialEq, Serialize)]
 #[serde(tag = "type")]
 pub enum TxFeeDetails {
-    Utxo(UtxoFeeDetails),
     Eth(EthTxFeeDetails),
     Qrc20(Qrc20FeeDetails),
     Slp(SlpFeeDetails),
     Tendermint(TendermintFeeDetails),
     #[cfg(all(not(target_os = "ios"), not(target_os = "android"), not(target_arch = "wasm32")))]
     Solana(SolanaFeeDetails),
+    Utxo(UtxoFeeDetails),
 }

 /// Deserialize the TxFeeDetails as an untagged enum.
@@ -981,21 +981,21 @@ impl<'de> Deserialize<'de> for TxFeeDetails {
         #[derive(Deserialize)]
         #[serde(untagged)]
         enum TxFeeDetailsUnTagged {
-            Utxo(UtxoFeeDetails),
             Eth(EthTxFeeDetails),
             Qrc20(Qrc20FeeDetails),
             #[cfg(all(not(target_os = "ios"), not(target_os = "android"), not(target_arch = "wasm32")))]
             Solana(SolanaFeeDetails),
             Tendermint(TendermintFeeDetails),
+            Utxo(UtxoFeeDetails),
         }

         match Deserialize::deserialize(deserializer)? {
-            TxFeeDetailsUnTagged::Utxo(f) => Ok(TxFeeDetails::Utxo(f)),
             TxFeeDetailsUnTagged::Eth(f) => Ok(TxFeeDetails::Eth(f)),
             TxFeeDetailsUnTagged::Qrc20(f) => Ok(TxFeeDetails::Qrc20(f)),
             #[cfg(all(not(target_os = "ios"), not(target_os = "android"), not(target_arch = "wasm32")))]
             TxFeeDetailsUnTagged::Solana(f) => Ok(TxFeeDetails::Solana(f)),
             TxFeeDetailsUnTagged::Tendermint(f) => Ok(TxFeeDetails::Tendermint(f)),
+            TxFeeDetailsUnTagged::Utxo(f) => Ok(TxFeeDetails::Utxo(f)),
         }
     }
 }
```

Then you will see type will show as `Solana` then. For now, I will remap it to `Tendermint` in `my_tx_history_v2` as temporary solution.

Reply:
I suppose it's because `TendermintFeeDetails` has the same fields names: `coin` and `amount`. It chooses the first successful variant, ignoring `gas_limit` field.